src/adhocracy_core/adhocracy_core/auditing/test_init.py

```diff
@@ -0,0 +1,173 @@
+import transaction
+import json
+
+from unittest.mock import Mock
+from pytest import fixture
+from pytest import mark
+from pyramid import testing
+
+
+@fixture()
+def integration(config):
+    config.include('adhocracy_core.content')
+    config.include('adhocracy_core.events')
+    config.include('adhocracy_core.changelog')
+    config.include('adhocracy_core.sheets')
+
+
+@fixture
+def user():
+    from adhocracy_core.sheets.principal import IUserBasic
+    user = testing.DummyResource(__provides__=IUserBasic)
+    user.name = 'god'
+    return user
+
+
+@fixture
+def mock_get_user_info(monkeypatch):
+    import adhocracy_core.auditing
+    mock_get_user_info = Mock(spec=adhocracy_core.auditing)
+    mock_get_user_info.return_value = ('god',
+                                       '/principals/users/000001')
+    monkeypatch.setattr(adhocracy_core.auditing,
+                        '_get_user_info',
+                        mock_get_user_info)
+    return mock_get_user_info
+
+
+def _get_event_name(index):
+    return 'event_{}'.format(index)
+
+
+def _get_payload_value1(index):
+    return 'value1_{}'.format(index)
+
+
+@fixture
+def context(context):
+    mocked_conn = Mock()
+    mocked_auditconn = Mock()
+    mocked_auditconn.root.return_value = {}
+    mocked_conn.get_connection.return_value = mocked_auditconn
+    context._p_jar = mocked_conn
+    return context
+
+
+@fixture()
+def request_(registry):
+    request = testing.DummyResource(registry=registry)
+    return request
+
+
+def test_add_events(context):
+    from . import log_auditevent
```

Comment:
Its a bit unclear what do want to test here and why do you slow down the test with 10000 entries?

A unit test should be named after the test condition or  the expected behavior.
The program code should follow the structure "given" "when", "then"
and test one "thing" (preferred one assert statement)

Also in the following test its not always clear what function is under test (get_auditlog or log_auditevent).